docs/source/send-recv.rst

```diff
@@ -0,0 +1,93 @@
+Send/Recv Internals
+===================
+
+Generally UCX creates connections between endpoints with the following steps:
+
+1. Create a ``Listener`` with defined IP address and port
+  a. ``Listener`` defines a callback function to process communications from endpoints
+2. Connect an ``Endpoint`` to the ``Listener``
+3. ``Endpoint`` communicates with the ``Listener``
```

Comment:
Maybe we should now add that the endpoint closes a connection, and eventually the listener too?

Reply:
Done